cl/recap/tasks.py

```diff
@@ -362,29 +343,29 @@ def process_recap_pdf(self, pk):
 
     if not pq.debug:
         try:
-            rd.save()
+            await rd.asave()
         except (IntegrityError, ValidationError):
             msg = "Duplicate key on unique_together constraint"
-            mark_pq_status(pq, msg, PROCESSING_STATUS.FAILED)
+            await mark_pq_status(pq, msg, PROCESSING_STATUS.FAILED)
             rd.filepath_local.delete(save=False)
             return None
 
     if not existing_document and not pq.debug:
-        extract_recap_pdf_base(rd.pk),
-        add_items_to_solr([rd.pk], "search.RECAPDocument")
+        await extract_recap_pdf_base(rd.pk),
+        await sync_to_async(add_items_to_solr)([rd.pk], "search.RECAPDocument")
```

Comment:
We should just do this via celery, because this is making an IO call. I'm baffled why we didn't before, so, perhaps:

```suggestion
        add_items_to_solr.delay([rd.pk], "search.RECAPDocument")
```

Or would it need to be:

```suggestion
        await sync_to_async(
            add_items_to_solr.delay([rd.pk], "search.RECAPDocument")
        )
```

Reply:
> We should just do this via celery, because this is making an IO call.

Um, we don't need celery for IO bound operations from my understanding, asyncio handles IO bound stuff fine(either natively or via fallback threadpool).

Reply:
Still, it'd be nice for the user not have to wait for it, right?

Reply:
I'm not sure what you mean, AFAIU we very much do want to wait for these operations to complete so we can then return an appropriate success/fail response code back to the client that made the original request. If we aren't able to appropriately notify the client if the request was successful then implementing functional retry logic in the client becomes nearly impossible. Since we are using ASGI there's very little overhead in just maintaining an open connection to the client while we wait for processing operations to complete.

Reply:
This is just adding it to the search engine. It'll pretty much always work, so I think it's safe to tell the user, "It worked!" even if the Solr part might still be happening somewhere.

Reply:
> It'll pretty much always work

I mean, being able to better detect edge case failures easily is kinda handy though.

Reply:
It'll show up via Sentry and the many other places this task is used.

Reply:
Sure, but I just don't see a reason to use celery here and prevent notifying the client of the failure as well(having the failure visible in more places doesn't seem to have downsides that I'm seeing), the `add_items_to_solr` function seems to be effectively doing entirely IO bound operations which shouldn't block the event loop. Down the line we should convert `add_items_to_solr` to a proper async function anyways(for now the threadpool based fallback should work fine).

Reply:
OK, if you say so. You've been right a lot lately. :)